app/src/ui/lib/path-text.tsx

```diff
@@ -374,13 +373,13 @@ export class PathText extends React.PureComponent<
     // by not measuring the actual width.
     if (availableWidth === 0) {
       if (this.state.length !== 0) {
-        this.setState({
-          ...this.state,
-          ...createPathDisplayState(this.state.normalizedPath, 0),
+        this.setState(prevState => ({
+          ...prevState,
```

Comment:
Do we need to pass in `prevState` here as well? I thought `setState` supported passing in just the values that need to be updated...

Reply:
You are correct, I have removed line 377 here.